src/npcmove.cpp

```diff
@@ -233,6 +233,16 @@ void npc::execute_action(npc_action action, int target)
                name.c_str(), target, npc_action_name(action).c_str());
     */
 
+    // Before doing anything else, if the player is friendly and we're
+    // blocking them, get out of the way.
+    if (attitude_to(g->u) == Creature::A_FRIENDLY && rl_dist(pos(), g->u.pos()) == 1) {
+        // We are only blocking movement if we're one tile away from the player.
+        if(is_blocking_position(g->u.pos())) {
+            move_away_from(g->u.pos());
+            return;
+        }
+    }
+
```

Comment:
Would it be worth it to make this more generic, to prevent them from blocking other NPCs as well?  This wouldn't affect the player as much and wouldn't be as annoying obviously, but there could be situations where its necessary.